x-pack/plugins/maps/public/classes/layers/tiled_vector_layer/tiled_vector_layer.tsx

```diff
@@ -54,9 +66,37 @@ export class TiledVectorLayer extends VectorLayer {
     this._source = source as ITiledSingleLayerVectorSource;
   }
 
-  getCustomIconAndTooltipContent() {
+  getMetaFromTiles(): TileMetaFeature[] {
+    return this._descriptor.__metaFromTiles || [];
+  }
+
+  getCustomIconAndTooltipContent(): CustomIconAndTooltipContent {
+    const tileMetas = this.getMetaFromTiles();
+    if (!tileMetas.length) {
+      return NO_RESULTS_ICON_AND_TOOLTIPCONTENT;
+    }
+
+    const totalFeatures: number = tileMetas.reduce((acc: number, tileMeta: Feature) => {
+      const count = tileMeta && tileMeta.properties ? tileMeta.properties[KBN_FEATURE_COUNT] : 0;
+      return count + acc;
+    }, 0);
+
+    if (totalFeatures === 0) {
+      return NO_RESULTS_ICON_AND_TOOLTIPCONTENT;
+    }
+
+    const isIncomplete: boolean = tileMetas.some((tileMeta: Feature) => {
+      return !tileMeta?.properties?.[KBN_IS_TILE_COMPLETE];
+    });
+
     return {
-      icon: <EuiIcon size="m" type={this.getLayerTypeIconName()} />,
+      icon: this.getCurrentStyle().getIcon(),
+      tooltipContent: isIncomplete
+        ? i18n.translate('xpack.maps.tiles.resultsTrimmedMsg', {
+            defaultMessage: `Layer shows incomplete results`,
```

Comment:
Can you change to "Results limited to {count} documents.". This will better match the message from ESSearchSource.